test/controllers/works_controller_test.rb

```diff
@@ -1,116 +1,288 @@
 require 'test_helper'
 
 describe WorksController do
-  describe "root" do
-    it "succeeds with all media types" do
-      # Precondition: there is at least one media of each category
-
+  let(:book) { works(:poodr) }
+
+  let(:user) { users(:dan) }
+
+  let(:bad_id) {Work.last.id + 1}
+
+  let(:work_data) {
+    work_data = {
+      work: {
+        category: 'books',
+        title: 'new work title',
+        publication_year: 1234
+      }
+    }
+  }
+
+  describe 'logged in user' do
+    describe "root" do
+      it "succeeds with all media types" do
+        # Precondition: there is at least one media of each category
+        perform_login(user)
+        get root_path
+        must_respond_with :success
+      end
+
+      it "succeeds with one media type absent" do
+        # Precondition: there is at least one media in two of the categories
+        perform_login(user)
+        book.destroy
+
+        get root_path
+        must_respond_with :success
+      end
+
+      it "succeeds with no media" do
+        perform_login(user)
+        Work.all.each do |work|
+          work.destroy
+        end
+
+        get root_path
+        must_respond_with :success
+      end
     end
 
-    it "succeeds with one media type absent" do
-      # Precondition: there is at least one media in two of the categories
-
+    CATEGORIES = %w(albums books movies)
+    INVALID_CATEGORIES = ["nope", "42", "", "  ", "albumstrailingtext"]
+
+    describe "index" do
+      it "succeeds when there are works" do
+        perform_login(user)
+        get users_path
```

Comment:
Since all of these tests require the user to be logged in, the `perform_login` might go well in a `before` block.